### ISSUE TITLE
Incorrect sourcemaps generated when use string template in function with no return type

### ISSUE THREAD

**Issue by pmunin** [Microsoft/TypeScript#18964]
I have a Typescript solution, where I also use Babel and Webpack. I use decorators in the my code. I found a problem that when I use an optional member decorator in class and somewhere after (in the same file) I use string template expression, then when I debug application Google Chrome shows me source as a STRING constrant from the point where the template expression starts till the end of a file. The code works through, it's just hard to debug. 

**TypeScript Version:**  2.5.3 / nightly (2.5.0-dev.201xxxxx)

**Code**

```ts
function myDecorator(...params):any
{
    return ;
}

export class MyClass
{
    @myDecorator()
    optionalProperty?:any;

}

function test(p1,p2,p3)
{
    let a= `${123}`
    let b = "1235345";
    let c = b + "asdas"
    debugger;
    console.log(c);
}

test(11,22,33)
```

**Expected behavior:**
proper display of code in google chrome dev tools

**Actual behavior:**
![image](https://user-images.githubusercontent.com/7120440/31234853-fa47d948-a9bd-11e7-8a8a-94868bf09757.png)

**Comment by weswigham** [comment]
@andy-ms a few weeks ago, said this was an issue with the code highlighter used by chrome. If that's the case, this complaint may be better directed at codemirror. @andy-ms is this the same issue you were seeing?

**Comment by weswigham** [comment]
100%, this is a chrome devtools issue with codemirror. The [latest codemirror](http://codemirror.net/mode/javascript/index.html) doesn't exhibit this behavior, but you can open the devtools, find any js file in the page and paste the above code into it to witness the behavior. In fact, [here's the crbug](https://bugs.chromium.org/p/chromium/issues/detail?id=675502&q=codemirror&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified). Watch that issue for when they mark it as closed when they update codemirror.
